### PR TITLE
[Needs review] Test and fix for issue #9326

### PR DESCRIPTION
**Pull request by toolforger**
Fixes issue #9326

### CODE REVIEW

**Comment by toolforger** [comment]
So far, `bin/test evalf` on Python 2.6 looks good.
Leaving in-depth testing to Travis.

File: sympy/core/tests/test_evalf.py [hunk 1]
```diff
@@ -472,3 +472,10 @@ def test_issue_8853():
     assert get_integer_part(S.Half, 1, {}, True) == (1, 0)
     assert get_integer_part(-S.Half, -1, {}, True) == (-1, 0)
     assert get_integer_part(-S.Half, 1, {}, True) == (0, 0)
+
+def test_issue_9326():
+    from sympy import Dummy
+    d1 = Dummy('d')
+    d2 = Dummy('d')
+    e = d1 + d2
+    assert e.evalf(subs = {d1: 1, d2: 2}) == 3
```
Comment (debugger22) [inline sympy/core/tests/test_evalf.py]: `assert e.evalf(subs={d1: 1, d2: 2})` and a newline.

File: sympy/core/tests/test_evalf.py [hunk 2]
```diff
@@ -472,3 +472,10 @@ def test_issue_8853():
     assert get_integer_part(S.Half, 1, {}, True) == (1, 0)
     assert get_integer_part(-S.Half, -1, {}, True) == (-1, 0)
     assert get_integer_part(-S.Half, 1, {}, True) == (0, 0)
+
+def test_issue_9326():
```
Comment (debugger22) [inline sympy/core/tests/test_evalf.py]: One more newline.

**Comment by toolforger** [comment]
Tests now pass (except for code_quality).
This still needs an in-depth review by someone who knows what potential ramifications this might have beyond those covered in testing ( @fredrik-johansson , @asmeurer , @certik , @moorepants ?)

**Comment by toolforger** [comment]
I just verified that the added test in commit #1 will indeed fail unless the fix in commit #2 is applied.

**Comment by moorepants** [comment]
I think this is fine. +1

I don't have any comments about ramifications from test cases that are not currently implemented. Unless you can think of test cases that are not present in the evalf tests, then I think this should go in.

**Comment by toolforger** [comment]
Issue is that I'd have to find, understand and validate all uses of `evalf` to be sure I didn't break anything. Such a small change, and it's potentially affecting almost all of SymPy - I wanted to see whether that raises an eyebrow for anybody who knows more about general coding practices in SymPy.
Given that nobody else has talked back in over a month, I guess it's okay.

I'll squash the codestyle commit, if the test doesn't sound an alarm bell it can go in.

**Comment by bjodah** [comment]
I've looked carefully and fail to see any side effects (the test suite is there for a reason after all).
I too am +1 to merge this, so I am merging. Thanks!

**Pull request merged**